### PR TITLE
revise security guidance

### PR DESCRIPTION
**Pull request by bcressey**
**Issue number:**
N/A

**Description of changes:**
Strengthen the warning against privileged containers, since they can bypass other protection mechanisms such as SELinux in various ways.

Upgrade the warning against sharing host namespaces, and clarify the additional risk of sharing the host PID namespace.

Expand the section on system mounts to cover local storage concerns.

Mention the new `control_t` label for API socket access, and the path to the Docker runtime socket at `/run/docker.sock`.

**Testing done:**
Rendered locally with `grip`.

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

File: SECURITY_GUIDANCE.md [hunk 1]
```diff
@@ -30,11 +30,36 @@ However, it is always better to patch vulnerabilities than to rely on mitigation
...
+Syscalls are a way for userspace programs to request services from the kernel.
+Seccomp filters can be used to allow access to a subset of syscalls.
+Bottlerocket uses `containerd` as the container runtime which provides [a default seccomp profile](https://github.com/containerd/containerd/blob/master/contrib/seccomp/seccomp_default.go).
+
+SELinux labels are part of mandatory access controls, which impose constraints after discretionary access controls are checked.
+Bottlerocket runs all containers with the unprivileged `container_t` label today.
+However, privileged containers may run with the privileged `super_t` label in the future.
+
+Orchestrators provide ways to disable these protections.
+For example, Docker can run containers with the `--privileged` flag, and Kubernetes can run pods with `privileged: true` set in the pod definition.
```
Comment (samuelkarp) [inline SECURITY_GUIDANCE.md]: Is it worth adding something about ECS here as well?  ECS allows containers to be run with a `privileged` flag in the task definition.

File: SECURITY_GUIDANCE.md [hunk 2]
```diff
@@ -106,16 +132,17 @@ The SELinux policy for Bottlerocket defines special transition rules for contain
 
 A container runtime can transition a child processes to either of these labels:
 * `container_t` (the default, for ordinary containers)
+* `control_t` (for containers that need to access the API)
 * `super_t` (for "superpowered" containers)
 
 Some orchestrators such as Kubernetes allow SELinux labels to be defined in the container specification.
```
Comment (samuelkarp) [inline SECURITY_GUIDANCE.md]: "such as Kubernetes or Amazon ECS"?

**Review by tjkirch — APPROVED** [review]

File: SECURITY_GUIDANCE.md [hunk 3]
```diff
@@ -106,16 +135,17 @@ The SELinux policy for Bottlerocket defines special transition rules for contain
 
 A container runtime can transition a child processes to either of these labels:
```
Comment (webern) [inline SECURITY_GUIDANCE.md]: Now more than two:

```suggestion
A container runtime can transition a child processes to any of these labels:
```

**Review by webern — APPROVED** [review]

**Review by samuelkarp — APPROVED** [review]

**Review by zmrow — APPROVED** [review]
⚡

**Review by jhaynes — APPROVED** [review]

**Pull request merged**